### PR TITLE
Use AwaitUntil() when running curl

### PR DESCRIPTION
**Pull request by vthapar**
When service is created/deleted, it can take a while to update
multiclusterservices objects for it across clusters.

Added retry to curl so we wait for curl to fail and not return
at initial success during delete use case.

### CODE REVIEW

File: test/e2e/dataplane/service_discovery.go [hunk 1]
```diff
@@ -122,19 +121,23 @@ func verifyClusterIpWithDig(f *framework.Framework, cluster framework.ClusterInd
...
+	framework.AwaitUntil("verify if service is discoverable", func() (interface{}, error) {
+		stdout, _, err := f.ExecWithOptions(framework.ExecOptions{
+			Command:       cmd,
+			Namespace:     f.Namespace,
+			PodName:       pod.Name,
+			ContainerName: pod.Spec.Containers[0].Name,
+			CaptureStdout: true,
+			CaptureStderr: true,
+		}, cluster)
+		if (err != nil && expectingSuccess) || (err == nil && !expectingSuccess) {
```
Comment (mkolesnik) [inline test/e2e/dataplane/service_discovery.go:136]: Won't this condition give a false positive?
`(err == nil && !expectingSuccess)`

 i.e. if err= nil then it'll return nil as the error, and youll get a success even though you were were expecting it to fail

Comment (vthapar) [reply test/e2e/dataplane/service_discovery.go:136]: > Won't this condition give a false positive?
> `(err == nil && !expectingSuccess)`
> 
> i.e. if err= nil then it'll return nil as the error, and youll get a success even though you were were expecting it to fail

Yes, will fix this.

**Comment by mangelajo** [comment]
I think you should at least change the commit to say "use retries" , AwaitUntil is an specific
function of the framework you could use for that :D

**Comment by vthapar** [comment]
> I think you should at least change the commit to say "use retries" , AwaitUntil is an specific
> function of the framework you could use for that :D

I did update the git commit msg and title, but github doesn't update the PR. will do.

File: test/e2e/dataplane/service_discovery.go [hunk 2]
```diff
@@ -122,19 +123,24 @@ func verifyClusterIpWithDig(f *framework.Framework, cluster framework.ClusterInd
...
+			Command:       cmd,
+			Namespace:     f.Namespace,
+			PodName:       pod.Name,
+			ContainerName: pod.Spec.Containers[0].Name,
+			CaptureStdout: true,
+			CaptureStderr: true,
+		}, cluster)
+		if err == nil && !expectingSuccess {
+			By(fmt.Sprintf("Retrying curl because got no error when expected to get one"))
+			time.Sleep(time.Second)
```
Comment (mkolesnik) [inline test/e2e/dataplane/service_discovery.go:140]: I don't think this sleep of a second matters much

**Pull request merged**